src/os/bluestore/bluestore_types.h

```diff
@@ -730,8 +731,10 @@ struct bluestore_blob_t {
     }
   }
 
-  int map(uint64_t x_off, uint64_t x_len,
-	   std::function<int(uint64_t,uint64_t)> f) const {
+  template<class F>
+  int map(uint64_t x_off, uint64_t x_len, F&& f) const {
+    static_assert(std::is_invocable_r_v<int, F, uint64_t, uint64_t>,
```

Comment:
nit, no need to have the error message of "F shall be invocable...", as the `static_assert()` is self-documenting, i think.

Reply:
TODO. [Since C+17 the message is optional](http://en.cppreference.com/w/cpp/language/static_assert). Thanks for pointing this out, @tchaikov!

Reply:
FIXED.